rosidl_typesupport_c/resource/srv__type_support.cpp.em

```diff
@@ -1,49 +1,65 @@
-// generated from rosidl_typesupport_c/resource/srv__type_support.cpp.em
-// generated code does not contain a copyright notice
-
-@#######################################################################
-@# EmPy template for generating <srv>__type_support.cpp files
-@#
-@# Context:
-@#  - spec (rosidl_parser.ServiceSpecification)
-@#    Parsed specification of the .srv file
-@#  - subfolder (string)
-@#    The subfolder / subnamespace of the message
-@#    Either 'srv' or 'action'
-@#  - type_supports (list of strings)
-@#    The name of the type support packages
-@#  - get_header_filename_from_msg_name (function)
-@#######################################################################
+@# Included from rosidl_typesupport_c/resource/idl__type_support.c.em
+@{
+TEMPLATE(
+    'msg__type_support.cpp.em',
+    package_name=package_name, interface_path=interface_path,
+    message=service.request_message, include_directives=include_directives,
+    type_supports=type_supports)
+}@
+
+@{
+TEMPLATE(
+    'msg__type_support.cpp.em',
+    package_name=package_name, interface_path=interface_path,
+    message=service.response_message, include_directives=include_directives,
+    type_supports=type_supports)
+}@
+
+@{
+from rosidl_cmake import convert_camel_case_to_lower_case_underscore
+include_parts = [package_name] + list(interface_path.parents[0].parts) + \
+    [convert_camel_case_to_lower_case_underscore(interface_path.stem)]
+include_base = '/'.join(include_parts)
+
+header_files = [
+    'cstddef',
+    'rosidl_generator_c/service_type_support_struct.h',
+    package_name + '/msg/rosidl_typesupport_c__visibility_control.h',
+]
+if len(type_supports) != 1:
+    header_files += [
+        'rosidl_typesupport_c/identifier.h',
+        'rosidl_typesupport_c/service_type_support_dispatch.h',
+        'rosidl_typesupport_c/type_support_map.h',
+    ]
+header_files.append('rosidl_typesupport_interface/macros.h')
```

Comment:
Hmm, this is the only place where we add this header unconditionally (every other place checks for `len(type_supports) != 1`). Why?

Reply:
Because that was exactly how it was before. So simply no change in semantic.

Could it be changed? Probably.

Reply:
That's a being pragmatic :) 

Fair enough.